src/arrayMethodReduce.js

```diff
@@ -1,11 +1,24 @@
 'use strict';
 
-/**
- * Implement method Reduce
- */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let accumulator;
+
+    if (arguments.length < 2) {
+      accumulator = this[0];
+
+      for (let i = 1; i < this.length; i++) {
+        accumulator = callback(accumulator, this[i], i, this);
+      }
+    } else {
+      accumulator = initialValue;
+
+      for (let i = 0; i < this.length; i++) {
+        accumulator = callback(accumulator, this[i], i, this);
+      }
+    }
```

Comment:
Do not repeat your self 
```suggestion
    let accumulator = initialValue;
    let index = 0
    
    if (arguments.length < 2) {
      accumulator = this[0];
      index = 1;
   }
   
    for (let i = index; i < this.length; i++) {
        accumulator = callback(accumulator, this[i], i, this);
      }
```